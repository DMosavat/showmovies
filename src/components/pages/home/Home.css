
.Home {
    max-width: 1000px;
    margin: auto;
    margin-top: 20px;
  }

.Home-img {
    max-height: 400px;
    max-width: 1100px;

  }