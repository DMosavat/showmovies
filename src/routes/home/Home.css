
.Home {
    max-width: 1000px;
    margin: auto;
    margin-top: 20px;
  }

.Home-noFav{
  margin-top: 100px;
  margin-bottom: 500px;
}
