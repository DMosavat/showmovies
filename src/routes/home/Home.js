
const apiKey = 'bf9a59578fcb278c84eca3877be1aed1'

export default {

    fetchApi(showType=0, mName= '') {

        return new Promise((resolve) => {

            let url = ''
            if( showType === 0 ){
                url = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}`
            }else if( showType === 1 ){
                url = `https://api.themoviedb.org/3/movie/top_rated?api_key=${apiKey}`
        
            }else{
                url = `https://api.themoviedb.org/3/search/movie?api_key=${apiKey}&query=${mName}`
        
            }

            let moviesTemp = []

            fetch(url).then((res) => res.json()).then((data) => {
        
                data.results.forEach(element => {
                    
                    if( element.vote_count > 0 ){
                        let movies = [ 
                        {
                            id: element.id,
                            title: element.title,
                            release_date: element.release_date,
                            vote_average: element.vote_average,
                            vote_count: element.vote_count,
                            poster_path: 'https://image.tmdb.org/t/p/w500' + element.poster_path
                        }
                        ]
                            
                        moviesTemp = [...moviesTemp, ...movies]
                    }
                })

            }).then(() => {
                resolve(moviesTemp)

            })
        })

    }
}



    
    
