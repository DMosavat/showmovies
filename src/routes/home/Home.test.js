import React from 'react';
import { shallow } from 'enzyme';
import Home from './index'

const apiKey = 'bf9a59578fcb278c84eca3877be1aed1'

describe('<Home />', () => {

    it('should render the form', () => {
        const wrapper = shallow(<Home />);
        expect(wrapper.html()).toMatchSnapshot();
    })

})


describe('check themoviedb path' , () => {

    it('it should get discover movies' ,  done => {
        fetch(`https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}`)
        .then(() =>{
            done()
        })
        .catch((err) => {
            console.log(err);
        })

    })

    it('it should get top movies' ,  done => {
        fetch(`https://api.themoviedb.org/3/movie/top_rated?api_key=${apiKey}`)
        .then(() =>{
            done()
        })
        .catch((err) => {
            console.log(err);
        })

    })

    it('it should can search movies' ,  done => {
        fetch(`https://api.themoviedb.org/3/search/movie?api_key=${apiKey}&query=jack`)
        .then(() =>{
            done()
        })
        .catch((err) => {
            console.log(err);
        })

    })


})