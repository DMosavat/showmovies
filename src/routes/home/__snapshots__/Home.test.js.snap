// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the form 1`] = `"<div class=\\"Home\\"><div><div class=\\"MovieSearch-Card card\\"><div class=\\"card-body\\"><div class=\\"row\\"><input type=\\"text\\" class=\\"MovieSearch-input\\" placeholder=\\"Search movie name ...\\" name=\\"mName\\" value=\\"\\"/><div><button class=\\"btn btn-link MovieSearch-button\\"> New Film</button><button class=\\"btn btn-link MovieSearch-button\\"> Top Film</button><button class=\\"btn btn-link MovieSearch-button\\"> Favorites</button></div></div></div></div></div><div><div class=\\"row justify-content-md-center\\"></div><div class=\\"Home-noFav\\"></div></div></div>"`;
